### PR TITLE
Fix LiteLLM key management and user migration SQL queries

### PR DESCRIPTION
**Pull request by tofarr**
## Summary of PR

This PR addresses two issues in the enterprise backend:

### 1. LiteLLM Key Cleanup on Entry Creation

When creating LiteLLM entries in cloud deployment mode, the system now explicitly deletes any existing key by alias before generating a new one. This handles environments where multiple OpenHands installations share the same Keycloak and LiteLLM instance, preventing stale keys from causing authentication issues.

**Changes:**
- Added `_delete_key_by_alias` call before `_generate_key` in `create_entries`
- Gracefully handles 404 errors when the key does not exist

### 2. User Migration SQL Query Fixes

Fixed SQL parameter binding issues in the user migration flow (`_migrate_personal_data`) where UUID and string parameters were being mixed incorrectly:

- Fixed `conversation_metadata` migration to properly handle UUID vs text user_id comparisons
- Fixed `stripe_customers`, `slack_users`, `slack_conversation`, `api_keys`, `custom_secrets`, and `billing_sessions` updates to use correct parameter types (UUID for org_id, string for user_id)

## Change Type

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Refactor
- [ ] Other (dependency update, docs, typo fixes, etc.)

## Checklist

- [x] I have read and reviewed the code and I understand what the code is doing.
- [x] I have tested the code to the best of my ability and ensured it works as expected.

## Release Notes

- [ ] Include this change in the Release Notes.

---

To run this PR locally, use the following command:

GUI with Docker:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.openhands.dev/openhands/runtime:118456f-nikolaik   --name openhands-app-118456f   docker.openhands.dev/openhands/openhands:118456f
```

### CODE REVIEW

File: enterprise/storage/user_store.py [hunk 1]
```diff
@@ -295,67 +295,66 @@ async def migrate_user(
...
-                        :user_id
+                        :user_uuid,
+                        :user_uuid
                     FROM conversation_metadata
-                    WHERE user_id = :user_id
+                    WHERE user_id = :user_id_text
                 """),
-                {'user_id': user_id},
+                {'user_uuid': user_uuid, 'user_id_text': user_id},
             )
```
[the remark this answers is not in the document]
Comment (tofarr) [reply enterprise/storage/user_store.py:312]: Fixed in 118456f6c. Added three comprehensive tests for the SQL parameter type handling:

1. `test_migrate_user_sql_type_handling` - Tests all 7 migration SQL statements with proper UUID vs string type handling. Verifies data is migrated correctly for conversation_metadata_saas, stripe_customers, slack_users, slack_conversation, api_keys, custom_secrets, and billing_sessions.

2. `test_migrate_user_sql_no_matching_records` - Tests edge case where no records match the user_id, verifying the migration doesn't fail.

3. `test_migrate_user_sql_multiple_conversations` - Tests migrating multiple conversations for a single user.

The tests set up legacy data with the old schema assumptions (string user_ids), execute the migration SQL, and verify the data migrated correctly with proper UUID/string types.
